src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -79,22 +85,26 @@ public class EventHubAdapterFactory : IQueueAdapterFactory, IQueueAdapter, IQueu
         /// <summary>
         /// Creates a message cache for an eventhub partition.
         /// </summary>
-        protected Func<string, IStreamQueueCheckpointer<string>, Logger, IEventHubQueueCache> CacheFactory { get; set; }
+        protected IEventHubQueueCacheFactory CacheFactory { get; set; }
```

Comment:
For compatability, please leave this as a func.
I'm fine with encapsulating the cache factory logic in it's own class that can be specialized and replaced, but the minimal behavior that needs replaces is this factory func, no reason to force developers to use our cache factory class or interface.